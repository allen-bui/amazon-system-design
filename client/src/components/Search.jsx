import React, { Component } from "react";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: "",
      searchResult: []
    };
    this.handleChange = this.handleChange.bind(this);
  }

  // updates the search value
  handleChange(e) {
    let query = e.target.value;
    console.log(query, `SEARCH VALUE`);
    this.setState({ search: query }, () => {
      let result = [...this.state.searchResult];
      console.log(result, `EMPTY RESULT?`);
      const { questions } = this.props;

      for (let i = 0; i < questions.length; i++) {
        let question = questions[i].question.split(" ");
        // console.log(question, `line 22`);
        // if the query is found in the question str
        if (
          question.includes(query) &&
          !result.includes(questions[i].question_id)
        ) {
          // insert the question into the result's array if it isn't already included
          console.log(questions[i], `question with query`);
          result.push(questions[i]);
        }
      }

      if (result.length) {
        this.setState({ searchResult: result });
        console.log(this.state, `result of search query`);
      }
    });
  }

  render() {
    return (
      <div className="a-section a-spacing-base askBtfSearchFormLabel askAutocomplete">
        <div className="a-search a-span12">
          <i className="fa fa-search a-icon a-icon-search" />
          <input
            type="search"
            maxLength="150"
            autoComplete="off"
            placeholder="Have a question? Search for answers"
            name="askQuestionText"
            className="a-input-text a-span12 askBtfSearchTextInput askBtfSearchPostTextInput askAutocompleteTextInput"
            value={this.state.search}
            onChange={e => this.handleChange(e)}
          />
        </div>
      </div>
    );
  }
}

export default Search;
