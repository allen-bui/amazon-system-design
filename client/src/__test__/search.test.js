import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Search from '../components/Search';

describe('Search Test Suite', () => {
  it('should mount in full DOM', () => {
    expect(mount(<Search/>).find('.askAutocomplete').length).toBe(1);
  });

  it('should render correclty with no props', () => {
    const component = shallow(<Search/>);
    expect(component).toMatchSnapshot();
  });
  it('should be selectable by class "a-spacing-base"', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper.is('.a-spacing-base')).toBe(true);
  });
});