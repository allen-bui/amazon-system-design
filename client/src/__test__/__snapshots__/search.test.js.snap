// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Test Suite should render correclty with no props 1`] = `ShallowWrapper {}`;
